client.go

```diff
@@ -446,6 +450,18 @@ func DialWithTLS(network, address string, tlsConfig *tls.Config) (conn *Conn, er
 	return conn, nil
 }
 
+// Client.DialTimeoutEx acts like DialTimeout, but uses the client's settings.
```

Comment:
Can't we just look at the value of LocalAddr and use it if is not nil? Think a new method is quite a big hammer for this feature.